vault/token_store.go

```diff
@@ -1013,42 +1074,69 @@ func (ts *TokenStore) lookupSalted(ctx context.Context, saltedID string, tainted
 
 // Revoke is used to invalidate a given token, any child tokens
 // will be orphaned.
-func (ts *TokenStore) revokeOrphan(ctx context.Context, id string) error {
+func (ts *TokenStore) revokeOrphan(ctx context.Context, tokenID string) error {
 	defer metrics.MeasureSince([]string{"token", "revoke"}, time.Now())
-	if id == "" {
+	if tokenID == "" {
 		return fmt.Errorf("cannot revoke blank token")
 	}
 
-	saltedID, err := ts.SaltID(ctx, id)
+	idHMAC, err := ts.hmac(ctx, tokenID)
```

Comment:
I think this block can be replaced with a `lookupTokenNonLocked` call

Reply:
`lookupTokenNonLocked` is not being used here because it doesn't return the `obfuscatedID` and we need that here.